pavelib/assets.py

```diff
@@ -20,75 +20,174 @@
-    """
-    if not systems:
-        systems = ALL_SYSTEMS
-    applicable_directories = []
-    applicable_directories.extend(COMMON_SASS_DIRECTORIES)
-    if "lms" in systems:
-        applicable_directories.extend(LMS_SASS_DIRECTORIES)
-    if "studio" in systems or "cms" in systems:
-        applicable_directories.extend(CMS_SASS_DIRECTORIES)
-    applicable_directories.extend(THEME_SASS_DIRECTORIES)
+
+# system specific lookup path additions, add sass dirs if one system depends on the sass files for other systems
+SASS_LOOKUP_DEPENDENCIES = {
+    'cms': [path('lms') / 'static' / 'sass' / 'partials', ],
+}
+
+THEME_SASS_WATCHER_DIRS = []
+
+
+def get_sass_directories(system, theme_dir):
+    """
+    Determine the set of SASS directories to be compiled for the specified list of system and theme
+    and return a list of those directories.
+
+    Each item in the list is dict object containing the following key-value pairs.
+    {
+        "sass_source_dir": "",  # directory where source sass files are present
+        "css_destination_dir": "",  # destination where css files would be placed
+        "lookup_paths": [],  # list of directories to be passed as lookup paths for @import resolution.
+    }
+
+    if theme_dir is empty or None then return sass directories for the given system only. (i.e. lms or cms)
+
+    :param system: name if the system for which to compile sass e.g. 'lms', 'cms'
+    :param theme_dir: absolute path of theme for which to compile sass files.
+    """
+    if system not in SYSTEMS:
+        raise ValueError("'system' must be one of ({allowed_values})".format(allowed_values=', '.join(SYSTEMS.keys())))
+    system = SYSTEMS[system]
+
+    applicable_directories = list()
+
+    # add common sass directories
+    applicable_directories.append({
+        "sass_source_dir": path("common/static/sass"),
+        "css_destination_dir": path("common/static/css"),
+        "lookup_paths": [
+            path("common/static"),
+            path("common/static/sass"),
+        ],
+    })
+
+    if theme_dir:
+        # Add theme sass directories
+        applicable_directories.extend(
+            get_theme_sass_dirs(system, theme_dir)
+        )
+    else:
+        # add system sass directories
+        applicable_directories.extend(
+            get_system_sass_dirs(system)
+        )
+
     return applicable_directories
 
 
+def get_theme_sass_dirs(system, theme_dir):
+    """
+    Return list of sass dirs that need to be compiled for the given theme. Also add theme sass directories to
+    THEME_SASS_WATCHER_DIRS variables so that they could be added to sass watcher later.
+
+    :param system: name if the system for which to compile sass e.g. 'lms', 'cms'
+    :param theme_dir: absolute path of theme for which to compile sass files.
+    """
+    if system not in ('lms', 'cms'):
+        raise ValueError('"system" must either be "lms" or "cms"')
+
+    dirs = []
+
+    system_sass = path(system) / "static" / "sass"
```

Comment:
Nit: can these three variables end in ``_dir`` to make it clear that they are directories.